source/session.ts

```diff
@@ -578,14 +582,18 @@ export class Session {
       pushToken,
       signal,
       additionalHeaders = {},
-      decodeDatesAsIso = false,
+      decodeDatesAsIso = undefined,
     }: CallOptions = {},
   ): Promise<IsTuple<T> extends true ? T : T[]> {
     if (this.initializing) {
       await this.initializing;
     }
     const url = `${this.serverUrl}${this.apiEndpoint}`;
 
+    decodeDatesAsIso = decodeDatesAsIso === undefined ?
+      this.decodeDatesAsIso :
+      decodeDatesAsIso;
+
```

Comment:
```suggestion
      decodeDatesAsIso = this.decodeDatesAsIso,
    }: CallOptions = {},
  ): Promise<IsTuple<T> extends true ? T : T[]> {
    if (this.initializing) {
      await this.initializing;
    }
    const url = `${this.serverUrl}${this.apiEndpoint}`;
```

I haven't tested it, but this way should work, unless I'm totally mistaken. You can also set the default to `this.decodeDatesAsIso` in the decode method as well.